File: src/SoftielRemote.Core/Dtos/ConnectionResponse.cs
using SoftielRemote.Core.Enums;

namespace SoftielRemote.Core.Dtos;

/// <summary>
/// Backend'in bağlantı isteğine verdiği yanıt.
/// </summary>
public class ConnectionResponse
{
    /// <summary>
    /// Bağlantı durumu.
    /// </summary>
    public ConnectionStatus Status { get; set; }

    /// <summary>
    /// İşlem başarılı mı?
    /// </summary>
    public bool Success { get; set; }

    /// <summary>
    /// Hata durumunda hata mesajı.
    /// </summary>
    public string? ErrorMessage { get; set; }

    /// <summary>
    /// Bağlantı kurulduğunda Agent'ın IP adresi ve port bilgisi (TCP bağlantısı için).
    /// </summary>
    public string? AgentEndpoint { get; set; }

    /// <summary>
    /// Bağlantı ID'si (birden fazla bağlantıyı takip etmek için).
    /// </summary>
    public string? ConnectionId { get; set; }
}

File: legacy/src/SoftielRemote.App.Flutter/lib/models/connection_response.dart
import 'connection_status.dart';

/// Connection response model
/// Backend API'den gelen bağlantı yanıtı
class ConnectionResponse {
  final bool success;
  final ConnectionStatus status;
  final String? errorMessage;
  final String? agentEndpoint;
  final String? connectionId;
  
  ConnectionResponse({
    required this.success,
    required this.status,
    this.errorMessage,
    this.agentEndpoint,
    this.connectionId,
  });
  
  factory ConnectionResponse.fromJson(Map<String, dynamic> json) {
    return ConnectionResponse(
      success: json['success'] as bool? ?? false,
      status: ConnectionStatusFromInt.fromInt(json['status'] as int? ?? 0),
      errorMessage: json['errorMessage'] as String?,
      agentEndpoint: json['agentEndpoint'] as String?,
      connectionId: json['connectionId'] as String?,
    );
  }
}

